core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -253,13 +356,17 @@ abstract class BaseConsumerTest extends IntegrationTestHarness {
   }
 
   protected class ConsumerAssignmentPoller(consumer: Consumer[Array[Byte], Array[Byte]],
-                                           topicsToSubscribe: List[String]) extends ShutdownableThread("daemon-consumer-assignment", false)
+                                           topicsToSubscribe: List[String],
+                                           partitionsToAssign: Set[TopicPartition] = Set.empty[TopicPartition]) extends ShutdownableThread("daemon-consumer-assignment", false)
```

Comment:
Seems like we should take either `topicsToSubscribe` or `partitionsToAssign`, but not both. Maybe we can have two separate constructors. Also, `subscribe` below should probably throw an illegal state error if `partitionsToAssign` is not empty.

Reply:
I added two new secondary constructors - can you check?